### PR TITLE
[kube-prometheus-stack] Add thanosRuler configuration

### PR DESCRIPTION
**Pull request by artificial-aidan**
#### What this PR does / why we need it:

Follows a similar pattern to `alertmanager` and allows configuration of a `ThanosRuler` instance and access.

#### Special notes for your reviewer:

I started by copying the whole alertmanager directory, but soon realized I would not be able to test all of those configurations. So I added what I can test to the best of my ability.

Mentions: @andrewgkew @desaintmartin @scottrigby @Xtigyro

#### Checklist

- [ X ] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ X ] Chart Version bumped
- [ X ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by artificial-aidan** [comment]
@monotek Is there anyone else I should tag on this for review/discussion?

**Comment by monotek** [comment]
One of the maintainers:
https://github.com/prometheus-community/helm-charts/blob/main/.github/CODEOWNERS#L10

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 34.10.0
+version: 34.11.0
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:26]:
```suggestion
version: 35.3.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by artificial-aidan** [comment]
Still waiting on approval

**Comment by monotek** [comment]
Still waiting that merge conflicts are solved.

File: charts/kube-prometheus-stack/templates/ruler/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  namespace: {{ template "kube-prometheus-stack.namespace" . }}
+  labels:
+    app: {{ template "kube-prometheus-stack.name" . }}-thanos-ruler
+{{ include "kube-prometheus-stack.labels" . | indent 4 }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "kube-prometheus-stack.name" . }}-thanos-ruler
+      release: {{ $.Release.Name | quote }}
+      self-monitor: "true"
```
Comment (monotek) [inline charts/kube-prometheus-stack/templates/ruler/servicemonitor.yaml:15]:
```suggestion
      self-monitor: {{ .Values.thanosRuler.serviceMonitor.selfMonitor | quote }}
```

File: charts/kube-prometheus-stack/templates/ruler/extrasecret.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.thanosRuler.extraSecret.data -}}
```
Comment (monotek) [inline charts/kube-prometheus-stack/templates/ruler/extrasecret.yaml:1]: can we rename the directory from "ruler" to "thanos-ruler" please?

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**